code/client/makepkginfo

```diff
@@ -385,6 +385,15 @@ def check_mode(option, opt, value, parser):
                                                             (opt, mode))
     setattr(parser.values, option.dest, value)
 
+def has_valid_install_critieria(catinfo):
+    """
+    Inspects compiled catalog info dictionary for valid install criteria.
+    Returns boolean.
+    """
+    results = []
+    for key in ['installs', 'receipts', 'installcheck_script']:
+        results.append(catinfo.has_key(key) and len(catinfo[key]) != 0)
+    any(results)
```

Comment:
While functionally it works, it implies something that's not true: that a PayloadIdentifier is used to detect installation status in the general case. At the very least I'd remove the mention of the PayloadIdentifier from the warning.

I'm not too worried about the case of a pkginfo for a configuration profile install that is missing a PayloadIdentifier, see https://github.com/munki/munki/blob/master/code/client/makepkginfo#L122-L126

You're just using the existence of the PayloadIdentifier key as a way to tell that the pkginfo is for a config profile install, and therefore we don't need to check for the existence of any of ['installs', 'receipts', 'installcheck_script']. 

One additional note about the warning -- you spelled "behavior" wrong. :-)

```python
print >> sys.stderr, (
            "WARNING: %s contains no discernible install criteria!\n"
            "If you do not supply a receipt, install item, or"
            "installcheck_script, \n"
            "Munki will have no way to determine \n"
            "the package's installation state. This can result in \n"
            "Munki repeatedly attempting to install this item, \n"
            "or failure to attempt install at all.") % item
```

I reserve the right to further tweak the warning message, but let's proceed with the patch.